### ISSUE TITLE
[GUIDE]: ORIGIN FIX

### ISSUE THREAD

**Issue by Crd5** [Whisky-App/Whisky#272]
# Guide on how to install Origin and bypass EA App installation.

### P.S. The guide is outdated but still should work. Ill improve it soon
 **UPD** You can use gptk with it if you disable hardware acceleration (After following the guide)

 **UPD** Since August, this can dont work. If you have issues, please comment

1) Create a new bottle

2) Enable DXVK in the bottle (Config > DXVK)

<img width="1346" alt="Screenshot 2023-07-30 at 13 37 16" src="https://github.com/IsaacMarovitz/Whisky/assets/62203291/8ecb9fdb-7e3a-44f0-b5d3-9ad6df9ec434">

3) Run Origin setup (Download it from [here](https://taskinoz.com/downloads/OriginSetup-10.5.115.51547.exe)

4) Install Origin as usual (DO NOT CHANGE PATH)

5) Log in

6) You will see the EA App installation request. Dont worry

7) Close Whisky (Command + Q)

8) Open Whisky and select your bottle

9) Download [this program](https://github.com/p0358/Fuck_off_EA_App/releases/tag/v2)

10) Run and install it (If you play Origin games via Steam, click "Pretend EA App is installed")

<img width="609" alt="Снимок экрана 2023-07-30 в 13 22 00" src="https://github.com/IsaacMarovitz/Whisky/assets/62203291/766ad673-d278-4ae7-ab9a-bde9f524d3fc">

11) Open wine configurator (Config > Open WIne Configurator), go to Libraries

12) Select "version" in "New override for library" and click "add"

<img width="522" alt="Screenshot 2023-07-30 at 13 31 48" src="https://github.com/IsaacMarovitz/Whisky/assets/62203291/be7d7ae3-27cb-47e5-bc52-307b3aa9e79f">

13) Select "version" in "Existing overrides" and click "Edit"

<img width="481" alt="Screenshot 2023-07-30 at 13 33 40" src="https://github.com/IsaacMarovitz/Whisky/assets/62203291/b45c9ba9-dfea-425d-923e-381f78faf4b4">

14) Select "Native then Builtin" if not selected and click "OK"

15) Click "Apply", "OK"

16) Click "Open C Drive"

17) Open C:\Program Files (X86)\Origin\EACore.ini in textedit app

18) Add:

[Bootstrap]
EnableUpdating=false

19) Save file (Command + S)

20) Open Origin

### If you have any issues, feel free to comment

**Comment by Crd5** [comment]
If you need help or have issues comment here I will try to help everyone

**Comment by Crd5** [comment]
@IsaacMarovitz pin this message please

**Comment by itkind** [comment]
Does this mean that all games will use DXVK and not GPTK ?

**Comment by Crd5** [comment]
> Does this mean that all games will use DXVK and not GPTK ?

Yes. Idk why but when I try to run Origin using gptk I get black screen. Soon Ill post a fix for EA App which will use gptk

**Comment by pallazola** [comment]
> > Does this mean that all games will use DXVK and not GPTK ?
> 
> Yes. Idk why but when I try to run Origin using gptk I get black screen. Soon Ill post a fix for EA App which will use gptk

Yes, I'm having the same black screen issue, looking forward to a new way, if the EA app can work

**Comment by Crd5** [comment]
But for some reason, it works in patched Crossover. Maybe it Whisky bug

**Comment by umberto-bianchin** [comment]
Hi, I follow this tutorial and origin works fine, then I installed Fifa (I tried more than one version) but when I click on the play button origin minimizes but nothing happen, after 5 seconds the button return "play" as it is when you haven't started the game

**Comment by 10-Gauge** [comment]
Any idea when the EA App fix will be coming?  Will this allow games that use EAC to work?  I am trying to get Battlefield 2042 running on my Mac.

**Comment by Dnblgn07** [comment]
hey man I got some problem I hope you can help I got an ea game from steam and I want to download it so I downloaded origin with steam (in same bottle ) but now steam doesnt works black screen

**Comment by hardcod3dd** [comment]
I cannot use this right now.Origin asks to update but fails,offline mode can't login.its a dead end

**Comment by Aeteriss** [comment]
Not sure if this is helpful as I'm not really even sure how much of this works, but none of this worked for me until I installed it into the same bottle as Steam and then launch all my origin games through Steam.

**Comment by hardcod3dd** [comment]
> Not sure if this is helpful as I'm not really even sure how much of this works, but none of this worked for me until I installed it into the same bottle as Steam and then launch all my origin games through Steam.

im installing all to steam bottle too but can't login to origin. origin needs update but then fails to update

**Comment by LeaveNhA** [comment]
Hey guys, 

I can tell you all the story to catch you up.
Easy Anti Cheat is a common anti-cheat software. For obvious reasons, the software needs to run itself in kernel level in Windows. With this, EAC needs to run kernel level in every other system/platform that you might wanna run it.
But EAC has a problem. EAC knows that every other OS has a different architecture, to make them different and unique.
So, I saw couple of things about EAC has an initiative to make it run under Linux but I don't trust them because I couldn't verify the knowledge, so, no resource, no hopes.

There is a Wine version for running EAC, but, I haven't tried it out, yet. I'll inform you guys if I manage to run it.

**Comment by lukewlms** [comment]
Followed the instructions but when I run It Takes Two through Steam it still runs EAAppInstaller and not Origin.

**Comment by Dnblgn07** [comment]
> If you need help or have issues comment here I will try to help everyone

link to download orign doesnt works now

**Comment by joshandersn** [comment]
Trying to launch Titanfall 2. It skips the EA installer, but it opens the EA Link app in a black window. I assume it's trying to link my steam account to my EA

**Comment by blasserre** [comment]
Any fixes to use GPTK?

**Comment by recSlayer** [comment]
> Any fixes to use GPTK with origin?
@blasserre @pallazola @Crd5 
Disabling hardware acceleration fixes the black screen issue [link to reddit tutorial](https://www.reddit.com/r/origin/comments/q8o9gv/disable_origin_client_hardware_acceleration/)

**Comment by Crd5** [comment]
> > Any fixes to use GPTK with origin?
> 
> @blasserre @pallazola @Crd5 
> Disabling hardware acceleration fixes the black screen issue [link to reddit tutorial](https://www.reddit.com/r/origin/comments/q8o9gv/disable_origin_client_hardware_acceleration/)

That's great. I'll modify the guide soon. Thanks a lot

**Comment by f022yo9** [comment]
Does any one can play Star Wars Jedi: Survivor in whisky?
I've tried dozens of time, but it runs and stops repeatably.

**Comment by Nonkoiz** [comment]
It still prompts me to enable EAapp in order to continue. I want to play in steam It takes two

**Comment by ramimuhammad** [comment]
hi,

i got this messege errore when i install the app

The program IGOProry64 exe has encountered a serious problem and needs to close. We are sorry for the inconvenience.

any helpe??

**Comment by srichr** [comment]
Hi, I'm unable to run the 'F**k off EA app installer'. I download it and click it and whisky opens as so. But clicking run does nothing.
<img width="943" alt="Screenshot 2024-02-01 at 16 33 35" src="https://github.com/Whisky-App/Whisky/assets/158497696/e37fc0ec-bb94-4313-ac78-6e557c95bc03">

**Comment by Crd5** [comment]
> Hi, I'm unable to run the 'F**k off EA app installer'. I download it and click it and whisky opens as so. But clicking run does nothing. <img alt="Screenshot 2024-02-01 at 16 33 35" width="943" src="https://private-user-images.githubusercontent.com/158497696/301576354-e37fc0ec-bb94-4313-ac78-6e557c95bc03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4OTE3NjAsIm5iZiI6MTcwNjg5MTQ2MCwicGF0aCI6Ii8xNTg0OTc2OTYvMzAxNTc2MzU0LWUzN2ZjMGVjLWJiOTQtNDMxMy1hYzc4LTZlNTU3Yzk1YmMwMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwMlQxNjMxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMDQ3YzMzNzU1MDdlYmQzMTc2Mjk3NmZiODVhY2M2OTVjMjk4ZThlNzczZTMwODU0NzhkMzU1Y2I2YmRjMWU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FmV8KRH6wgOql8JupBOYz4GfkWVzG3hk6nwrh1MWxDo">

Thats strange that you have this window. Probably it was added recently. I would downgrade the Whisky

**Comment by recSlayer** [comment]
> hi,
> 
> i got this messege errore when i install the app
> 
> The program IGOProry64 exe has encountered a serious problem and needs to close. We are sorry for the inconvenience.
> 
> any helpe??
@ramimuhammad 

Ensure your bottle is shutdown and navigate to your bottle in finder and head to:

/drive_c/users/crossover/AppData/Roaming/Origin
Look for a file named "local_<hash>.xml" and open it. Change this line:
`  Setting key="EnableIgo" type="1" value="true"
`
to:
`Setting key="EnableIgo" type="1" value="false"
`

**Comment by recSlayer** [comment]
> > Any fixes to use GPTK with origin?
> 
> @blasserre @pallazola @Crd5 Disabling hardware acceleration fixes the black screen issue [link to reddit tutorial](https://www.reddit.com/r/origin/comments/q8o9gv/disable_origin_client_hardware_acceleration/)

Setting WINEDLLOVERRIDES="libglesv2=d" also ends up disabling acceleration and thus also fixes the black screen issue. You can do this through the wine cfg window in bottle configuration.

**Comment by Crd5** [comment]
> > > Any fixes to use GPTK with origin?
> > 
> > @blasserre @pallazola @Crd5 Disabling hardware acceleration fixes the black screen issue [link to reddit tutorial](https://www.reddit.com/r/origin/comments/q8o9gv/disable_origin_client_hardware_acceleration/)
> 
> Setting WINEDLLOVERRIDES="libglesv2=d" also ends up disabling acceleration and thus also fixes the black screen issue. You can do this through the wine cfg window in bottle configuration.

What is libglesv2 responsible for?

**Comment by captain0305** [comment]
<img width="808" alt="image" src="https://github.com/Whisky-App/Whisky/assets/63387697/fdd93e5a-029d-4812-a36d-4b7f585e8345">
I met this problem. any one can help? Ea Easy anticheat can't work

**Comment by blackandcold** [comment]
Worked to start Origin but can't start Mass Effect Andromeda - instantly vanishing without a message. Anyone still trying to make this work?

**Comment by IsaacMarovitz** [comment]
EA App now works in WhiskyWine 2.1.1 with the `d3dcompiler_47` trick.

**Comment by blackandcold** [comment]
well, now installer is stuck on "finishing install" and dxvk can't be used.
I'll check and create a new issue. Maybe we can get rid of the Origin hack-around

**Comment by Nonkoiz** [comment]
can't see EAapp login in UI, EACefSubProcess.exe error

Unhandled exception: page fault on read access to 0xffffffffffffffff in 64-bit code (0x000000033a7c55).
Register dump:
 rip:00000000033a7c55 rsp:000000000011c320 rbp:000000000011c520 eflags:00000202 (   - --  I   - - - )
 rax:0000000000000001 rbx:0000000004d38d90 rcx:0000000000000007 rdx:0000000000000000
 rsi:0000000003499e6d rdi:000000000011c398  r8:000000000011c320  r9:000000000011c328 r10:00000000002430b0
 r11:000000000011c318 r12:0000000004dcd3e0 r13:0000000000000000 r14:000000000011c460 r15:0000000000000004
Stack dump:
0x0000000011c320:  000000000011c398 0000000000000003
0x0000000011c330:  0000000000000003 000000000011c398
0x0000000011c340:  0000000003499e6d 00000000033f9ddb
0x0000000011c350:  0000000004d38d90 000000000011c448
0x0000000011c360:  000000000011c460 0000000004dcd3e0
0x0000000011c370:  000000000011c398 000000000011c448
0x0000000011c380:  0000000010012570 0000000002f8eaf9
0x0000000011c390:  aaaaaaaaaaaaaaaa 0000000003499e34
0x0000000011c3a0:  0000000000000591 0000000003499e27
0x0000000011c3b0:  0000000003499e0a 000000000011c460
0x0000000011c3c0:  ffffffff8ac5ab64 0000000000000009
0x0000000011c3d0:  0000567bc715b5b6 000000000011c448
Backtrace:
=>0 0x000000033a7c55 in libglesv2 (+0x4b7c55) (0x0000000011c520)
  1 0x000000033f9ddb in libglesv2 (+0x509ddb) (0x0000000011c520)
  2 0x00000002f8eaf9 ??0PlatformMethods@angle@@QEAA@XZ+0x9be69() in libglesv2 (0x0000000011c520)
  3 0x00000002f4fe31 ??0PlatformMethods@angle@@QEAA@XZ+0x5d1a1() in libglesv2 (0x0000000011c520)
  4 0x000001870acfaf CrashForExceptionInNonABICompliantCodeRange+0x1cdb4af() in libcef (0x0000000011c520)
  5 0x00000181f5e039 GetHandleVerifier+0x1264f9() in libcef (0x00000000000001)
  6 0x00000181f5d699 GetHandleVerifier+0x125b59() in libcef (0x00136c00011400)
  7 0x00000181f577f9 GetHandleVerifier+0x11fcb9() in libcef (0x00000000000001)
  8 0x00000181f66bea GetHandleVerifier+0x12f0aa() in libcef (0x00000000000040)
  9 0x0000018191ef20 IsCrashReportingEnabledImpl+0x15d4ab0() in libcef (0x0000000011cda8)
  10 0x0000018414ddf8 GetHandleVerifier+0x23162b8() in libcef (0x0000000011cda8)
  11 0x0000018191f1a1 IsCrashReportingEnabledImpl+0x15d4d31() in libcef (0000000000000000)
  12 0x0000018191e350 IsCrashReportingEnabledImpl+0x15d3ee0() in libcef (0x00136c00018098)
  13 0x0000018194e745 IsCrashReportingEnabledImpl+0x16042d5() in libcef (0x00136c00018098)
  14 0x00000182e1e47e GetHandleVerifier+0xfe693e() in libcef (0x00136c00018098)
  15 0x00000182e1aff9 GetHandleVerifier+0xfe34b9() in libcef (0x0000000000001d)
  16 0x000001843c1660 GetHandleVerifier+0x2589b20() in libcef (0x0000000000001d)
  17 0x00000182e1c70c GetHandleVerifier+0xfe4bcc() in libcef (0x00000004bb0234)
  18 0x000001820ec98e GetHandleVerifier+0x2b4e4e() in libcef (0x00000004bb0234)
  19 0x00000181ad3638 IsCrashReportingEnabledImpl+0x17891c8() in libcef (0x000000000000aa)
  20 0x00000181ad3249 IsCrashReportingEnabledImpl+0x1788dd9() in libcef (0000000000000000)
  21 0x000001820edd02 GetHandleVerifier+0x2b61c2() in libcef (0000000000000000)
  22 0x00000180e04a28 IsCrashReportingEnabledImpl+0xaba5b8() in libcef (0000000000000000)
  23 0x00000181a3bee7 IsCrashReportingEnabledImpl+0x16f1a77() in libcef (0xaaaaaaaaaaaaaaaa)
  24 0x000001840e7910 GetHandleVerifier+0x22afdd0() in libcef (0xaaaaaaaaaaaaaaaa)
  25 0x0000018419e75a GetHandleVerifier+0x2366c1a() in libcef (0xaaaaaaaaaaaaaaaa)
  26 0x0000018419ce35 GetHandleVerifier+0x23652f5() in libcef (0x000000000a8001)
  27 0x00000181e333b5 IsCrashReportingEnabledImpl+0x1ae8f45() in libcef (0x000000000a8001)
  28 0x00000181771d7e IsCrashReportingEnabledImpl+0x142790e() in libcef (0x000000000a8001)
  29 0x00000181d7e570 IsCrashReportingEnabledImpl+0x1a34100() in libcef (0x000000000a8001)
  30 0x00000181720282 IsCrashReportingEnabledImpl+0x13d5e12() in libcef (0x000000ffffffff)
  31 0x00000181720f94 IsCrashReportingEnabledImpl+0x13d6b24() in libcef (0x000000ffffffff)
  32 0x0000018171f8c5 IsCrashReportingEnabledImpl+0x13d5455() in libcef (0x000000ffffffff)
  33 0x0000018171f9df IsCrashReportingEnabledImpl+0x13d556f() in libcef (0x00000000000001)
  34 0x000001802bdc56 cef_add_cross_origin_whitelist_entry+0xcbfa6() in libcef (0x00000000000001)
  35 0x00000180291d19 cef_add_cross_origin_whitelist_entry+0xa0069() in libcef (0000000000000000)
  36 0x000001801f1442 cef_execute_process+0xa2() in libcef (0000000000000000)
  37 0x000001400037f8 in eacefsubprocess (+0x37f8) (0000000000000000)
  38 0x00000140001875 in eacefsubprocess (+0x1875) (0000000000000000)
  39 0x000001400398c6 in eacefsubprocess (+0x398c6) (0000000000000000)
  40 0x0000007b62ff19 ActivateActCtx+0x2430d() in kernel32 (0000000000000000)
  41 0x0000017006bb78 A_SHAFinal+0x495e8() in ntdll (0000000000000000)
0x000000033a7c55 libglesv2+0x4b7c55: int	$0x29
Modules:
Module	Address					Debug info	Name (76 modules)
PE	0000000000470000-0000000000673000	Deferred        dxgi
PE	0000000002ef0000-000000000361f000	Export          libglesv2
PE	0000000003620000-000000000369e000	Deferred        libegl
PE	00000000036a0000-00000000039a1000	Deferred        d3d11
PE	0000000010000000-0000000010002000	Deferred        usp10
PE	0000000068840000-0000000068843000	Deferred        win32u
PE	0000000068e30000-0000000068e34000	Deferred        opengl32
PE	000000006b150000-000000006b156000	Deferred        winemac
PE	000000007b000000-000000007b26a000	Deferred        kernelbase
PE	000000007b600000-000000007b66b000	Dwarf-5         kernel32
PE	0000000140000000-000000014006a000	Export          eacefsubprocess
PE	0000000170000000-00000001700b8000	Dwarf-5         ntdll
PE	0000000180000000-000000018bdda000	Export          libcef
PE	00000001c4ee0000-00000001c4f1b000	Deferred        winspool
PE	00000001c69e0000-00000001c732f000	Deferred        shell32
PE	00000001c8890000-00000001c88d5000	Deferred        d3dcompiler_47
PE	00000001c8b40000-00000001c8b6d000	Deferred        msacm32
PE	00000001c8db0000-00000001c8e6b000	Deferred        msvcrt
PE	00000001cfdf0000-00000001cfe13000	Deferred        dxva2
PE	00000001dd3f0000-00000001dd4d7000	Deferred        crypt32
PE	00000001ec2b0000-00000001ec2d8000	Deferred        ws2_32
PE	00000001f51e0000-00000001f51f2000	Deferred        hid
PE	00000001fdfd0000-00000001fdffd000	Deferred        wintrust
PE	0000000201cf0000-0000000201d12000	Deferred        secur32
PE	000000020fc60000-000000020fccc000	Deferred        mf
PE	000000021ff70000-0000000220220000	Deferred        wined3d
PE	0000000228450000-000000022847d000	Deferred        propsys
PE	0000000231ae0000-0000000231b76000	Deferred        rpcrt4
PE	000000023d820000-000000023da18000	Deferred        user32
PE	0000000240030000-000000024006a000	Deferred        iphlpapi
PE	000000024f470000-000000024f490000	Deferred        mpr
PE	00000002522e0000-000000025232d000	Deferred        oleacc
PE	000000025d740000-000000025d74e000	Deferred        dwmapi
PE	000000026b4c0000-000000026b555000	Deferred        gdi32
PE	000000026ed50000-000000026ed6b000	Deferred        vcruntime140
PE	00000002739c0000-0000000273b04000	Deferred        oleaut32
PE	0000000274a50000-0000000274a5e000	Deferred        kerberos
PE	000000028dfa0000-000000028dfad000	Deferred        nsi
PE	00000002967b0000-0000000296a84000	Deferred        d3d9
PE	000000029cfc0000-000000029cfdd000	Deferred        dnsapi
PE	00000002a4090000-00000002a409e000	Deferred        wtsapi32
PE	00000002bb750000-00000002bb8b0000	Deferred        comctl32
PE	00000002cc880000-00000002cc88c000	Deferred        wevtapi
PE	00000002d23f0000-00000002d241f000	Deferred        credui
PE	00000002d4d40000-00000002d4d59000	Deferred        bcrypt
PE	00000002e3540000-00000002e35a0000	Deferred        shlwapi
PE	00000002e8f10000-00000002e903e000	Deferred        ole32
PE	00000002f1fa0000-00000002f1fae000	Deferred        version
PE	00000002f3920000-00000002f3930000	Deferred        esent
PE	00000002f7230000-00000002f7270000	Deferred        uxtheme
PE	00000002f7f20000-00000002f7f2c000	Deferred        vcruntime140_1
PE	000000030a950000-000000030a9d0000	Deferred        dwrite
PE	00000003126f0000-000000031270b000	Deferred        shcore
PE	000000031f800000-000000031f905000	Deferred        comdlg32
PE	0000000327020000-0000000327079000	Deferred        combase
PE	000000032a700000-000000032a72d000	Deferred        sechost
PE	0000000330260000-00000003302a3000	Deferred        advapi32
PE	0000000330a70000-0000000330a82000	Deferred        uiautomationcore
PE	0000000335260000-0000000335273000	Deferred        rtworkq
PE	00000003422e0000-0000000342389000	Deferred        urlmon
PE	0000000346ba0000-0000000346bc1000	Deferred        jsproxy
PE	0000000358120000-0000000358171000	Deferred        winhttp
PE	0000000360e10000-0000000360e9e000	Deferred        msvcp140
PE	00000003673d0000-00000003673ff000	Deferred        netapi32
PE	00000003871c0000-000000038720e000	Deferred        concrt140
PE	0000000388e20000-0000000388e35000	Deferred        userenv
PE	00000003a0440000-00000003a04d7000	Deferred        wininet
PE	00000003a19e0000-00000003a19f0000	Deferred        ncrypt
PE	00000003a37f0000-00000003a37fd000	Deferred        dcomp
PE	00000003a6d40000-00000003a6da6000	Deferred        mfplat
PE	00000003aa4c0000-00000003aa4cc000	Deferred        dhcpcsvc
PE	00000003af670000-00000003af76c000	Deferred        ucrtbase
PE	00000003afd00000-00000003afd1b000	Deferred        imm32
PE	00000003b8f00000-00000003b8fd0000	Deferred        winmm
PE	00000003bb250000-00000003bb35d000	Deferred        cryptui
PE	00000003be590000-00000003be621000	Deferred        dbghelp
Threads:
process  tid      prio    name (all IDs are in hex)
00000030 services.exe
	00000034    0     
	00000038    0     
	00000044    0     
	0000005c    0     
	000000b4    0     
	00000104    0     
	00000124    0     
	00000970    0     
	0000079c    0     
0000003c svchost.exe
	00000040    0     
	0000004c    0     
	00000050    0     
00000054 winedevice.exe
	00000058    0     
	00000060    0     
	00000064    0     
	00000068    0     
	0000006c    0     
	00000118    0     
000000ac winedevice.exe
	000000b0    0     
	000000b8    0     
	000000bc    0     
	000000c0    0     
	000000c4    0     
	000000c8    0     
	000000cc    0     
000000f0 plugplay.exe
	000000f4    0     
	0000010c    0     
	00000110    0     
	00000114    0     
	0000018c    0     
	00000190    0     
	00000248    0     
	0000024c    0     
	00000648    0     
	00000770    0     
	00000774    0     
0000011c rpcss.exe
	00000120    0     
	00000128    0     
	0000012c    0     
	00000130    0     
	00000134    0     
	00000178    0     
	000003ac    0     
	000003cc    0     
	00000980    0     
	00000764    0     
	00000984    0     
00000160 steam.exe
	00000164    0     
	0000017c    0     
	00000180    0     
	00000184    0     
	00000188    0     
	0000019c    0     
	000001a0    0     
	000001a8    0     
	000001b0    0     
	000001b8    0     
	000001c0    0     
	000001d0    0     
	000001d4    0     
	000001d8    0     
	000001dc    2     
	000001e0    2     
	000001e8    0     
	000001ec    0     
	000001f0    0     
	00000260    0     
	00000264    0     
	00000268    0     
	00000274    0     
	00000278    2     
	0000028c    0     
	00000298    0     
	00000368    0     
	0000036c    0     
	00000378    0     
	00000380    0     
	00000388    0     
	0000038c    0     
	000003b8    0     
	000003bc    0     
	000003d4    0     
	000003e0    0     
	000003fc    0     
	00000444    0     
	000007dc    0     
	00000894    0     
	00000898    0     
	00000804    0     
00000168 explorer.exe
	0000016c    0     
	00000170    0     
	00000174    0     
00000194 steamwebhelper.exe
	00000198    0     
	000001ac    0     
	000001b4    0     
	000001f8    0     
	000001fc    0     
	00000204   -2     
	00000208    0     
	0000020c    0     
	00000210    0     
	00000214    0     
	00000218    0     
	0000021c    0     
	00000220    0     
	00000224    0     
	0000022c    0     
	00000230    0     
	00000234    0     
	00000238    0     
	0000023c    1     
	00000240    0     
	00000244    0     
	00000250   -2     
	00000254    0     
	00000258    0     
	0000027c    0     
	00000288    0     
	0000029c    0     
	000003a8    0     
00000280 steamwebhelper.exe
	00000284    0     
	000002a8    0     
	000002ac    0     
	000002b0    0     
	000002b4   -2     
	000002bc    0     
00000290 steamwebhelper.exe
	00000294    0     
	000002c0    0     
	000002c4    0     
	000002cc   -2     
	000002d0    0     
	000002d4    0     
	000002d8    0     
	000002dc    0     
	00000354    0     
000002a0 steamwebhelper.exe
	000002a4    0     
	000002f0    0     
	000002f4    0     
	000002fc   -2     
	00000304    1     
	00000308    0     
	0000030c    0     
	00000310    0     
	00000314    1     
	00000318    0     
	0000031c    0     
	00000320    0     
	00000324   -2     
	00000330    0     
	000003c4    0     
	00000528    0     
	0000052c   15     
	00000540    0     
	000006b8    0     
	0000058c    0     
0000044c steamwebhelper.exe
	00000450    0     
	00000460    0     
	00000468    0     
	00000470   -2     
	00000474    1     
	0000047c    0     
	00000480    0     
	00000484    0     
	00000488    1     
	0000048c    0     
	00000494    0     
	00000498    0     
	0000049c   -2     
	00000a88    0     
00000500 steamwebhelper.exe
	00000504    0     
	00000508    0     
	0000050c    0     
	00000510    0     
	00000514   -2     
	00000518    0     
	00000520    0     
	00000524    0     
	00000b50    0     
00000968 EABackgroundService.exe
	0000096c    0     
	00000978    0     
	0000097c    0     
	00000988    0     
	000009a4    0     
	000009a8    0     
	000009ac    0     
	000009b0    0     
	000009b4    0     
	000009b8    0     
	000009bc    0     
	000009c0    0     
	000009c4    0     
	000009c8    0     
	000009cc    0     
	000009d0    0     
	000009d4    0     
	000009d8    0     
	000009dc    0     
	000009e0    0     
	000009f0    0     
	000009f4    0     
	00000a10    0     
	00000a14    0     
	00000a64  -15     
	00000a68    0     
	00000a6c    0     
	00000a70    0     
	00000a74    0     
	00000a78    0     
	00000a7c    0     
	00000a80    0     
	00000814    0     
0000086c EADesktop.exe
	00000870    0     
	0000090c    0     
	000008cc    0     
	000008d8    0     
	00000904    0     
	000008e0    0     
	000008e4    0     
	00000908    0     
	000008f0    0     
	000008f4    0     
	000008f8    0     
	000008fc    0     
	00000900    0     
	00000910    0     
	00000918    0     
	00000914    0     
	000008c4    0     
	000008ac    0     
	00000938    0     
	0000093c    0     
	00000920    0     
	00000954    0     
	00000a50    0     
	00000a54    0     
	00000a4c   -2     
	00000a40    0     
	00000a3c    0     
	0000073c    0     
	000007d0    0     
	000007f0    0     
	000007c4    0     
	0000074c    0     
	00000750    0     
	0000075c    0     
	00000768    0     
	0000076c    0     
	00000778   -2     
	0000077c    0     
	000007a0    0     
	000007a4    0     
	000007a8    0     
	000007ac  -15     
	000007b4    2     
	000007b8    2     
	000007bc    0     
	00000724    0     
	00000720    0     
	0000072c    0     
	00000960    0     
	0000069c    0     
	00000698    0     
	000006a8    0     
	00000a60    0     
	000006bc    0     
	000006c8    0     
	000006cc    0     
	00000b68    0     
	00000b74    0     
	00000b98    0     
	00000bc4    0     
	00000be4    0     
	00000bf0    0     
	00000bec    0     
	00000bf8    0     
	0000043c    0     
	00000538    0     
	00000760   -2     
	00000b10    0     
	00000b84    0     
	00000b8c    0     
	00000bb8   15     
000007ec (D) C:\Program Files\Electronic Arts\EA Desktop\EA Desktop\EACefSubProcess.exe
	00000794    1 <== 
	000006d8    0     
	000006e4    0     
	0000070c    0     
	00000710    0     
	00000714   -2     
	00000690    1     
	00000680    0     
	0000068c    0     
	00000aa0    0     
	00000aac    1     
	00000328    0     
	0000032c    0     
	00000340    0     
	00000338    0     
	0000014c   -2     
	00000a2c    0     
000006ac EALocalHostSvc.exe
	000006b0    0     
	00000ab8    0     
	00000ab0    0     
	00000aa8    0     
	00000ab4    0     
	00000aa4    0     
	00000a98    0     
	00000a94    0     
	00000ad0    0     
	00000acc    0     
	00000ad8    0     
	00000ad4    0     
	00000ae0    0     
	00000adc    0     
	00000ae8    0     
	00000af4    0     
	00000b00    0     
	00000b14    0     
	00000b1c    0     
	00000b28    0     
	00000b24    0     
	00000b30    0     
	00000b2c    0     
	00000b34    2     
	00000b40    2     
	00000b3c    0     
	00000b4c    0     
	00000b58    0     
	00000b54    0     
00000b64 EACefSubProcess.exe
	00000b70    0     
	00000bd0    0     
	00000bcc    0     
	00000bd8    0     
	00000bd4   -2     
	00000bdc    0     
00000b80 EACefSubProcess.exe
	00000b7c    0     
	00000c00    0     
	00000bfc    0     
	00000c04    0     
	00000c10    0     
	00000c0c    0     
	00000c18    0     
	00000c1c   -2     
	0000013c    0     
	00000b78    0     
	00000bb4   -2     
00000b94 EACefSubProcess.exe
	00000ba0    0     
	00000080    0     
	00000088    0     
	000000a8    0     
	00000070   -2     
	00000090    1     
	00000084    0     
	0000009c    0     
	00000098    0     
	000001c8    1     
	000001cc    0     
00000ba4 EACefSubProcess.exe
	00000bb0    0     
	00000138    0     
	00000028    0     
	0000002c    0     
	00000154   -2     
	00000144    1     
	00000024    0     
	000001f4    0     
	00000a84    0     
	00000350    1     
	0000034c    0     
	000003c0    0     
	00000408    0     
	00000370   -2     
	00000b6c    0     
	00000b88    0     
0000099c conhost.exe
	00000a30    0     
00000748 svchost.exe
	00000a1c    0     
	00000730    0     
	000007c0    0     
	00000974    0     
	000000d0    0     
System information:
    Wine build: wine-7.7
    Platform: x86_64
    Version: Windows 10
    Host system: Darwin
    Host version: 23.1.0

**Comment by lukewlms** [comment]
> EA App now works in WhiskyWine 2.1.1 with the `d3dcompiler_47` trick.

Works for me now!  Specific steps:

1. New bottle
2. Click "Winetricks..." button
3. DLL tab -> run "d3dcompiler_47"

A terminal window opens and runs some stuff in the background.  I left the terminal window open, not sure if that's necessary.

**Comment by IsaacMarovitz** [comment]
@Nonkoiz you need to install the `d3dcompiler_47` Winetrick.

**Comment by Felipe-9** [comment]
With the `d3dcompiler_47` how the original guide changes? do we still need to install `FOffEA`? everything else is still needed?

**Comment by Nonkoiz** [comment]
> > EA App now works in WhiskyWine 2.1.1 with the `d3dcompiler_47` trick.
> 
> Works for me now! Specific steps:
> 
> 1. New bottle
> 2. Click "Winetricks..." button
> 3. DLL tab -> run "d3dcompiler_47"
> 
> A terminal window opens and runs some stuff in the background. I left the terminal window open, not sure if that's necessary.

now, I got this error：
Background services crashed
The background services aren't responding, sowe're kicking off the update manually. Stay putwhile we get the app up and ready for you.

**Comment by Nonkoiz** [comment]
Maybe it’s because the EAapp updates automatically? When launching the game on Steam, install EAapp will always be executed, causing the original version to be updated. Then I encountered the current crash problem, and "It takes two" cannot be played at all. I have no idea until now.

**Comment by Nonkoiz** [comment]
> > > > EA App now works in WhiskyWine 2.1.1 with the `d3dcompiler_47` trick.
> > > 
> > > 
> > > Works for me now! Specific steps:
> > > 
> > > 1. New bottle
> > > 2. Click "Winetricks..." button
> > > 3. DLL tab -> run "d3dcompiler_47"
> > > 
> > > A terminal window opens and runs some stuff in the background. I left the terminal window open, not sure if that's necessary.
> > 
> > 
> > now, I got this error： Background services crashed The background services aren't responding, sowe're kicking off the update manually. Stay putwhile we get the app up and ready for you.
> 
> Same issue. Do you have any solution?

Maybe it’s because the EAapp updates automatically? When launching the game on Steam, install EAapp will always be executed, causing the original version to be updated. Then I encountered the current crash problem, and "It takes two" cannot be played at all. I have no idea until now.

**Comment by MisterRnobe** [comment]
Hey guys
After installing `d3dcompiler_47` and EA App I managed to download `SWBF2` but after downloading it suggests to update the game. I relaunch EA App and now it does not display library games and shows banner with `things are taking a little longer than usual to load up...`. I also tried to launch BF2 directy by `starwarsbattlefrontii.exe`, it shows 'Preparing game' window with infinite loading process. Has anyone faced with same problem?

**Comment by yavda1** [comment]
Hi, when I try to run something it doesn't work and just closes. 

https://github.com/Whisky-App/Whisky/assets/94653933/8620fb3e-f5c3-4f02-9021-57c0e433eb40

Still using the origin method, the d3dcompiler_47 doesn't work for me.

**Comment by crimist** [comment]
I'm able to successfully install EA App with `d3dcompiler_47` but am not able to install battlefield 4. The game downloads fine but hangs infinitely on "preparing game" after installing all the dependancies (punkbuster, vcredist, etc).

Using Origin as defined above seems to mitigate the issue for anyone else with a similar problem.

**Comment by Spidy123222** [comment]
> I'm able to successfully install EA App with `d3dcompiler_47` but am not able to install battlefield 4. The game downloads fine but hangs infinitely on "preparing game" after installing all the dependancies (punkbuster, vcredist, etc).
> 
> Using Origin as defined above seems to mitigate the issue for anyone else with a similar problem.

unfortunately trying to get origin for the origin method just has a blank screen. launching ea app just doesn't launch as it tries to launch link2ea.

**Comment by feed0** [comment]
I can't play BF4 on Steam.

**Comment by BenjaOliva** [comment]
Hi, i tried this guide with no luck.

On my end im trying to run BF4 on Mac Air M2, Whisky v2.3.2.

When i try to run EA App it closes unexpectedly after ~2 min. So i can open the game even load the campaign and see the performance (All good there!) but suddenly the game closes and shows a EA App dialog about an error asking for reboot the app.

I tried all the Whisky bottle combinations, and having DXVK is a must since it wont open the game at all after 6 secs of loading screen and freeze. EA App not crashing if not using DXVK.

Yesterday (i think it was luck) I could play 8hs straight on DXVK no EA crashing an in all modes for BF4. For that, I logged out and log in again and the game session where perfectly normal. Not sure if i did something different or if was due to the many past tries that changed something lol. (Also, tried the EA Repair option for the game a couple times, didn't worked)

I also ran the Punkbuster installer manually by the `run` whisky bottle option.

Not sure whats the issue here, but im sure of a couple things:
- Game performs well, not a graphic issue. Good FPS and game experience
- Installation seems ok
- EA App takes ~2 min before crashing, every time. It opens a EA Updater icon so maybe thats the issue here, i assume
- DXVK seems to cause this issue (?) not so sure about this

Hope this helps to find a solution for latest state of this

**Comment by samarutuk** [comment]
The Origin app, i.e. starting games via **Origin, will be a thing of the past from April 17!** You will then have to use the EA app. I used this workaround because I couldn't get the EA app to work in Whisky. If this works for anyone else, I'd be curious to know how it works. According to WineHQ (https://appdb.winehq.org/objectManager.php?sClass=application&iId=20348) the EA app should work. Maybe it also depends on the Wine version, Whisky doesn't have the latest one.

**Comment by blackandcold** [comment]
@samarutuk you need to use DXVK to install and start EA.
Games, if they work (Mass Effect Andromeda for me), they work best with D3DMetal.
So start the EA App with DXVK and then switch to D3D Metal before starting the game.

EA App glitches hard with Metal API.

I don't think there was more to this.

**Comment by chozillla** [comment]
I actually was able to get this running! Mac Mini M4.

**Comment by sa1am8** [comment]
> I actually was able to get this running! Mac Mini M4.

Could you please provide Whisky configuration you are running on? Do you launch app through Steam, Origin or EA app? What packages you had to install to run it?

**Comment by chozillla** [comment]
> > I actually was able to get this running! Mac Mini M4.
> 
> [@chozillla](https://github.com/chozillla)
> 
> Could you please provide Whisky configuration you are running on? Do you launch app through Steam, Origin or EA app? What packages you had to install to run it?

<img width="895" alt="Image" src="https://github.com/user-attachments/assets/a469041f-6679-4796-a56a-fce5db8c7a33" /> BF4 runs smoothly

**Comment by chozillla** [comment]
<img width="702" alt="Image" src="https://github.com/user-attachments/assets/ae9dc73f-04a2-4a79-ae02-e74c518c4c74" />

**Comment by chozillla** [comment]
> > I actually was able to get this running! Mac Mini M4.
> 
> [@chozillla](https://github.com/chozillla)
> 
> Could you please provide Whisky configuration you are running on? Do you launch app through Steam, Origin or EA app? What packages you had to install to run it?

Origin and use the fuckEA plugin. Then I run the BF4.exe with Whisky

**Comment by jacktheporsche** [comment]
man that's amazing, hopefully I can get this running on my M2 pro
